core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -507,6 +515,7 @@ public String toString() {
             if (failure != null) {
                 components.add("failure [" + ExceptionsHelper.detailedMessage(failure) + "]");
             }
+            components.add("markAsStale [" + markAsStale + "]");
```

Comment:
it's a bit odd that shard started now also has this component. Maybe a good time to separate ShardEntry into StartedShardEntry and FailedShardEntry?

Reply:
while doing that, you can also convert that class to use final fields by not implementing `readFrom` but instead creating a constructor that takes StreamInput as parameter.